### PR TITLE
Natasha/youtube api task

### PR DESCRIPTION
**Pull request by Natasha08**
### Checklist:
- [x] Test your work and double check you didn't break anything
- [ ] Repeat steps to enable youtube api access on production

### Description:
[Related Issue](https://github.com/OriginProtocol/origin-website/issues/311)
- Provide routes for a user to authorize the website to access the api
- save access and refresh tokens to the db
- Add to the weekly job to check for youtube stats

### CODE REVIEW

**Comment by joshfraser** [comment]
I'm confused. Why do we need to create a new DB table for storing YouTube credentials? Can't we just put them in environment variables like everything else?

**Comment by Natasha08** [comment]
Someone has to get to go the url /youtube and authorize the site. The code
are then supplied in the redirect to obtain the access_token and the
refresh_token. Because of this when an individual does this we store the
tokens in the db. I suppose someone could go through those steps and then
update the environment variables manually. I’m not sure which is the best
approach.

On Feb 4, 2019, at 6:44 PM, Josh Fraser <notifications@github.com> wrote:

I'm confused. Why do we need to create a new DB table for storing YouTube
credentials? Can't we just put them in environment variables like
everything else?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/OriginProtocol/origin-website/pull/342#issuecomment-460483409>,
or mute the thread

**Comment by joshfraser** [comment]
I haven't checked, but you can usually get a long-lived API key. At a minimum, we can go through this process one time and store the resulting token as an env variable instead of creating a new DB table for it.

File: views/web_views.py [hunk 1]
```diff
@@ -270,6 +276,75 @@ def serve_origin_js(version):
...
+  flow.redirect_uri = url_for('oauth2callback', _external=True)
+
+  authorization_response = request.url
+  flow.fetch_token(authorization_response=authorization_response)
+
+  #save the token and refresh_token in the credentials as environment variables
+  credentials = flow.credentials
+
+  print("YOUTUBE TOKEN", credentials.token)
+  print("YOUTUBE REFRESH_TOKEN", credentials.refresh_token)
```
Comment (Natasha08) [inline views/web_views.py]: @joshfraser is printing to the console like this okay (to then throw into env vars)?

**Comment by Natasha08** [comment]
@joshfraser @tomlinton If there are no objections, this is ready for meege

**Comment by tomlinton** [comment]
It looks like it has migrations for both creating and then destroying the table that holds the Youtube credentials. I'll remove those.

**Comment by Natasha08** [comment]
@tomlinton Thanks I did not think of that

**Comment by tomlinton** [comment]
Tested this out live and we have Youtube stats coming in, thanks @Natasha08!

**Pull request merged**